test/value-of.ts

```diff
@@ -0,0 +1,47 @@
+﻿import { AssertTrue as Assert, IsExact } from "conditional-type-checks";
+import { ValueOf, Primitive, AnyArray } from "../lib/types";
+
+declare const array: number[];
+declare const func: (...arg: any[]) => boolean;
+declare const obj: { propA: string; propB: number; propC: () => void };
```

Comment:
Can you please add 2 other tests with readonly and optional properties?

* existing obj will be `writableObj`
* `optionalObj` with optional properties
* `readonlyObj` with readonly properties

e.g. like in Playground – https://tsplay.dev/mAre1W

Reply:
Sure!